include/soci/type-conversion.h

```diff
@@ -183,6 +183,21 @@ class conversion_use_type
     SOCI_NOT_COPYABLE(conversion_use_type)
 };
 
+template <typename T>
+class by_value_conversion_use_type : value_holder<T>, public conversion_use_type<T>
```

Comment:
It would be really great to have at least some explanation about what this class is used for.